pkg/sql/rename_table.go

```diff
@@ -286,25 +286,25 @@ func (p *planner) dependentFunctionError(
 	return sqlerrors.NewDependentBlocksOpError(op, typeName, objName, "function", fnDesc.GetName())
 }
 
-func (p *planner) dependentViewError(
+func (p *planner) dependentRelationError(
 	ctx context.Context,
 	typeName, objName string,
 	parentID descpb.ID,
-	viewDesc catalog.TableDescriptor,
+	desc catalog.TableDescriptor,
 	op string,
 ) error {
-	viewName := viewDesc.GetName()
-	if viewDesc.GetParentID() != parentID {
-		viewFQName, err := p.getQualifiedTableName(ctx, viewDesc)
+	viewName := desc.GetName()
+	if desc.GetParentID() != parentID {
+		viewFQName, err := p.getQualifiedTableName(ctx, desc)
 		if err != nil {
-			log.Warningf(ctx, "unable to retrieve name of view %d: %v", viewDesc.GetID(), err)
+			log.Warningf(ctx, "unable to retrieve name of relation %d: %v", desc.GetID(), err)
 			return sqlerrors.NewDependentObjectErrorf(
-				"cannot %s %s %q because a view depends on it",
-				op, typeName, objName)
+				"cannot %s %s %q because a %q depends on it",
+				op, typeName, objName, desc.GetObjectTypeString())
 		}
 		viewName = viewFQName.FQString()
 	}
-	return sqlerrors.NewDependentBlocksOpError(op, typeName, objName, "view", viewName)
+	return sqlerrors.NewDependentBlocksOpError(op, typeName, objName, desc.GetObjectTypeString(), viewName)
```

Comment:
super nit: we can improve the `NewDependentBlocksOpError` function by making the `dependentType` parameter be of type `redact.SafeString`

Reply:
Good suggestion. I believe `op` and `typeName` don't need to be redacted either. I opted to cast all three to `redact.SafeString` inside `NewDependentBlocksOpError` when formatting the message.